grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -205,29 +211,50 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
                 private HttpData content = HttpData.empty();
                 @Nullable
                 private HttpHeaders trailers;
+                @Nullable
+                private Subscription subscription;
+                @Nullable
+                private Throwable cause;
                 private int processedMessages;
 
                 @Override
                 public void onSubscribe(Subscription s) {
+                    if (subscription != null) {
+                        s.cancel();
+                        return;
+                    }
+                    subscription = s;
                     // At least 2 requests are required for receiving trailers.
                     s.request(2);
```

Comment:
We may just want to request `Long.MAX_VALUE` because we error if we receive more than one deframed non-trailer message anyway.

Reply:
What'd be the benefit of doing that? AFAIU requesting too much demand may result in producer allocating too many deframed messages that will have to be discarded (e.g. if for malformed response, or accidental use of streamed response)

Reply:
It's OK to do so because we abort the response anyway when we received more than expected, but yeah, it doesn't make much difference indeed.